scala3doc/src/dotty/renderers/HtmlRenderer.scala

```diff
@@ -0,0 +1,215 @@
+package dotty.dokka
+package renderers
+
+import HTML._
+import collection.JavaConverters._
+import java.net.URI
+import java.net.URL
+import dotty.dokka.model.api._
+import dotty.dokka.site._
+import scala.util.Try
+import org.jsoup.Jsoup
+import java.nio.file.Paths
+import java.nio.file.Path
+import java.nio.file.Files
+import java.nio.file.FileVisitOption
+import java.io.File
+
+case class Page(link: Link, content: Member | ResolvedTemplate | String, children: Seq[Page]):
+  def withNewChildren(newChildren: Seq[Page]) = copy(children = children ++ newChildren)
+
+  def withTitle(newTitle: String) = copy(link = link.copy(name = newTitle))
+
+  def hasFrame = content match
+    case t: ResolvedTemplate => t.hasFrame
+    case _ => true
+
+class HtmlRenderer(rootPackage: Member, val members: Map[DRI, Member])(using ctx: DocContext)
+  extends SiteRenderer, Resources, Locations, Writer:
+  private val args = summon[DocContext].args
+  val staticSite = summon[DocContext].staticSiteContext
+
+  private def memberPage(member: Member): Page =
+    val childrenPages = member
+      .membersBy(m => m.kind == Kind.Package || m.kind.isInstanceOf[Classlike])
+      .filter(m => m.origin == Origin.RegularlyDefined && m.inheritedFrom.isEmpty)
```

Comment:
There was a function that decided whether to render page for member or not (Kind.Package and Kind.Classlike can be nested in e.g. Kind.Given). Without checking nested kinds we don't generate pages for all members that should have page.

It should look like
```Scala
    def filterFunc(kind: Kind): Boolean = kind match {
      case Kind.Package => true
      case _ if kind.isInstanceOf[Classlike] => true
      case Kind.Given(inner, _, _) => filterFunc(inner)
      case Kind.EnumCase(inner) => filterFunc(inner)
      case _ => false
    }
    val childrenPages = member
      .membersBy(m => filterFunc(m.kind))
      .filter(m => m.origin == Origin.RegularlyDefined && m.inheritedFrom.isEmpty)
```